ops/model.py

```diff
@@ -254,6 +335,9 @@ def __init__(self, relation_name, relation_id, is_peer, our_unit, backend, cache
     def __repr__(self):
         return f'<{type(self).__module__}.{type(self).__name__} {self.name}:{self.id}>'
 
+    def __hash__(self):
+        return hash((self.id, self.name))
```

Comment:
I believe the ID itself is unique, but needs double checking.

I'm also curious about why this was required. We hopefully shouldn't have more than a single object representing the same relation inside the same logic. If that happens, we'll have other bugs even if we clash them like this.

Reply:
As far as I can tell from the code and operating experience, they are.

Both of the below:

`AddRelation`: https://github.com/juju/juju/blob/juju-2.7.2/state/state.go#L2208-L2215
`addPeerRelationsOps`: https://github.com/juju/juju/blob/juju-2.7.2/state/state.go#L1090

use `sequence` to get a new relation ID by incrementing a counter:
https://github.com/juju/juju/blob/juju-2.7.2/state/sequence.go#L21-L43

I dropped this in https://github.com/canonical/operator/pull/109/commits/010057b55a3674a1c30f5411d6fe767af6031fbe